packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -203,19 +319,43 @@ class HttpRouteSpec extends RouteSpec {
    */
   public readonly weightedTargets: WeightedTarget[];
 
+  /**
+   * The retry policy
+   */
+  public readonly retryPolicy?: HttpRetryPolicy;
+
   constructor(props: HttpRouteSpecOptions, protocol: Protocol) {
     super();
     this.protocol = protocol;
     this.match = props.match;
     this.weightedTargets = props.weightedTargets;
     this.timeout = props.timeout;
+
+    if (props.retryPolicy) {
+      if (props.retryPolicy.httpRetryEvents && props.retryPolicy.httpRetryEvents.length === 0) {
+        throw new Error('Specify at least one value in `httpRetryEvents` or leave it undefined');
+      }
+
+      if (props.retryPolicy.tcpRetryEvents && props.retryPolicy.tcpRetryEvents.length === 0) {
+        throw new Error('Specify at least one value in `tcpRetryEvents` or leave it undefined');
+      }
```

Comment:
Remove this validation, and simply return `undefined` if a customer provided an empty list, in `renderHttpRetryPolicy`.

No reason to burden the customer with unnecessary errors if we know what to do.

Reply:
No problem.